mayavi/action/filters.py

```diff
@@ -50,6 +45,7 @@ def perform(self, event):
         mv.add_filter(obj)
         mv.engine.current_selection = obj
 
+    @on_trait_change("mayavi:engine.[current_selection,current_object]")
```

Comment:
This change is what is causing https://github.com/enthought/mayavi/pull/999#issuecomment-754935361

The change was motivated due to a timing issue with trying to access `self.mayavi`.  Without the change, the app can not even be launched and we have an `AttributeError: 'NoneType' object has no attribute 'on_trait_change'`.

Reply:
This Action is being instantiated before the GUI is started.  Thus the handler `_on_application_gui_started` in `app.py` has not yet been called, so `script.engine` is not yet set.  

We need that link up to happen earlier (ie before these actions are instantiated), or we need this notification listener to get set up later, after that sync up happens (ie after the gui has started).